js/pulse.js

```diff
@@ -904,6 +907,10 @@
                 }
             };
 
+            var _onDeviceIdSet = function(event, deviceId) {
+                this._persistentId = deviceId
```

Comment:
I will add the check here. We still need the check I added in loadMetadata because otherwise the persistend id will be set to null if it did not exist in the metadata.